lib/maths/CTreeShapFeatureImportance.cc

```diff
@@ -367,10 +367,11 @@ const CTreeShapFeatureImportance::TStrVec& CTreeShapFeatureImportance::columnNam
     return m_ColumnNames;
 }
 
-double CTreeShapFeatureImportance::baseline(std::size_t classIdx) const {
-    double result{0.0};
+CTreeShapFeatureImportance::TVector CTreeShapFeatureImportance::baseline() const {
+
+    TVector result{las::zero((*m_Forest)[0][0].value())};
     for (const auto& tree : *m_Forest) {
-        result += tree[0].value()(classIdx);
+        result += tree[0].value();
```

Comment:
I think this warrants an explanation, i.e. tree[0] is the root and we compute internal node values by a weighted average of their leaf values.

Reply:
See [9d35bb2](https://github.com/elastic/ml-cpp/pull/1522/commits/9d35bb24e743761f526837f37661158666fccfc4).